Doc link to view url params (#393)
vgteam/sequenceTubeMap: public/help/help.md

4. If simplifying the BED file chunk or graph is possible, users will see a "Simplify Off" button, which when clicked with toggle to "Simplify On". This option enables vg simplify, which would remove small snarls. This option will only appear when there aren't any reads to be displayed. 
5. Click Go to see the selected tracks render in the visualization area.
![Go Button](helpGuideImages/img9.png)

##### How to make link-to-view URLs

> Users can compose URLs that link to a specific view. To do so, users will require specific information about files and tracks. The link has a [query string](https://en.wikipedia.org/wiki/Query_string#Structure) that consists of key-value pairs separated by ampersands, where the keys are not URL-econded and the values are URL-encoded. Keys can use brackets to encode hierarchical structures such as arrays and objects. For arrays, put a number in brackets to assign a new value to or access a value in that entry in the array, and for objects, put the key's name in brackets to assign a new value to or access a value in that entry in the object. The key's name should **not** be in quotes.

Reviewer: I get a little lost reading about the nuances of encoding hierarchical structures right off here, before I have an idea of why I would need to encode one.

I would paragraph-break after the second sentence here.

I would also maybe replace that second sentence with something more explanatory of the concept of a view; right now you're just saying that I will need "specific" (but unspecified) information about some unspecified files and tracks. You might want to say that the link can describe which tracks and files will be displayed, or something.

The details of the brackets should maybe move down to the description of `tracks` so it can be near its examples and so that it can come after we've introduced the problem of encoding an object with some JSON representation as part of a query string. Or, we could leave it here but provide some examples here and maybe give it a little section heading: "Query String Encoding" or something.

These are the fields that can be included in the URL:
1. `tracks` 

>    Information about tracks. Tracks are objects consisting of trackFile, trackType, and trackColorSettings. To retrieve this information, index the tracks array to access the object and respective keys.

Reviewer: The field names here should be in backticks. And I think we should cut the sentence about "retrieving" the information, because that doesn't make any sense in the context of *encoding* information into a link.

>    A track JSON object might look like this:

Reviewer: We should call this "a track object in JSON format"; we're about to convert it to a different format as part of the query string.

   ```
   {
      "trackFile": "exampleData/internal/snp1kg-BRCA1.vg.xg",
      "trackType": "graph",
      "trackColorSettings": {
         "mainPalette": "greys",
         "auxPalette": "ygreys",
         "colorReadsByMappingQuality": false
      }
   }
   ```

> 

Reviewer: Now we need to explain (or call back to our previous explanation) about how we're going to encode the collection of tracks as a bunch of query string parameters that all start with `tracks`.

   * The `trackFile` is the path (from the server working directory) or URL (any HTTP/HTTPS URL) to the track file 
     Examples:
      * `tracks[0][trackFile]=exampleData/internal/snp1kg-BRCA1.vg.xg`
      * `tracks[0][trackFile]=https://public.gi.ucsc.edu/~anovak/graphs/trivial-brca1.vg`

>    * The `trackType` specifies the type of the track, which can be "graph", "haplotype", "read"

Reviewer: We enclose the premade palette names in backticks, so we should probably use backticks for the enumeration values here.

      * Example: `tracks[0][trackType]=graph`

>    * The `trackColorSettings` provides color information and is an optional setting.
>       * The color palettes are `mainPalette` and `auxPalette`. 
>          * The colors for the palettes can be hex codes starting with a `#`, or premade palettes: `greys`, `ygreys`, `blues`, `reds`, `plainColors`, or `lightColors`. 
>          * The palettes are used differently for different track types

Reviewer: All these sub-bullets for the child fields should probably follow the same field-colon-phrase pattern as the top-level bullets. And we want to make sure all the bullets are punctuated consistently; they probably should all end with a period.

And instead of saying in text that something is optional we can always include it in parentheses so it's easy to see at a glance.

```suggestion
   * `trackColorSettings` (optional): the color information for the track. It has subfields:
      * `mainPalette` and `auxPalette` (optional): The "main" and "auxilliary" color palettes to use, respectively. The values for each palette can be a hex code starting with a `#`, or one of the premade palette names: `greys`, `ygreys`, `blues`, `reds`, `plainColors`, or `lightColors`. 
      The palettes are used differently for different track types:
```

            * For graphs, the `mainPalette` colors the primary reference path while `auxPalette` is used for the other paths. 
            * For haplotypes, only the `mainPalette` is used. 

>             - For reads, the `mainPalette` colors the forward-strand reads and the `auxPalette` colors the backward-strand reads. 

Reviewer: ```suggestion
            * For reads, the `mainPalette` colors the forward-strand reads and the `auxPalette` colors the reverse-strand reads. 
```

      * `colorReadsByMappingQuality` is a boolean value that determines if reads are colored based on their mapping quality or not. It is an optional field that has a default of `false`. 
      * Examples
         * `tracks[0][trackColorSettings][mainPalette]=greys`

>          * `tracks[0][trackColorSettings][auxPalette]=ygreys`

Reviewer: These examples should be moved under the bullet that represents both `mainPalette` and `auxPalette`, leaving the `colorReadsByMappingQuality` example under the bullet for that field.

>          * `colorReadsByMappingQuality=true`

Reviewer: ```suggestion
         * Example: `tracks[0][trackColorSettings][colorReadsByMappingQuality]=true`
```

2. `region`

>    This is a region input that is documented at step 3 of

Reviewer: I would call the data type here "region coordinates" or "a region specification" or even "a region string", rather than "a region input".

   [Displaying Visualizations](#displaying-visualizations). This region will be loaded in the tubemap visualization once the link is followed.

>    * Example: `region=17:3A1-100`

Reviewer: The URL encoding here is slightly wrong.
```suggestion
   * Example: `region=17%3A1-100`
```
Around here might also be a good place to remind people that the value needs to be URL-encoded, since it likely contains a colon.

3. `bedfile`

>    The `bedFile` is the path (from the server working directory) or URL (any HTTP/HTTPS URL) to the bed file. It is an optional field. More information on bedfile structure and creation is documented [here](https://github.com/vgteam/sequenceTubeMap?tab=readme-ov-file#preparing-subgraphs-in-advance). 

Reviewer: I don't know if we can talk about "the" bed file; we should introduce the concept here. We want to say that it is something like "a BED file containing coordinates for regions of interest, some of which may have associated pre-extracted Tube Map views", and link to the URL there to explain the extended BED file format we use.

We also want to do "(optional)" here if we want to switch from prose to that.

   * Examples:

>       * `bedFile=exampleData/internal/snp1kg-BRCA1.bed`

Reviewer: Does it have to be `bedFile` or `bedfile`? It is case sensitive I think so only one of these can be right.

      * `bedfile=https://raw.githubusercontent.com/vgteam/sequenceTubeMap/ca4f2485231ee4182173bec19489ba940b27461a/exampleData/cactus.bed`

>       * `bedfile=none`

Reviewer: We should note that this has special semantics; it means no BED file, not the BED file in the file named `none`.

> 4. `datatype`

Reviewer: I think this might really need to be `dataType`, with the capital T. That's what it is in the link example used in the tests.

   Describes type of data as `built-in`, `mounted files`, or synthetic `examples`.

>       * `built-in`: If the `datatype` field is set to `built-in`, you should set the `name` field to the name of a preset defined in `DATA_SOURCES` in `config.json`.

Reviewer: But what does this really mean, though? We should explain the value like we explain the fields, with a noun phrase before we explain additional semantics or give advice.

Also, maybe we should call them "data sources" and not "presets", since the config file does.

```suggestion
      * `built-in`: Predefined data sources from the server configuration file. If the `datatype` field is set to `built-in`, you should set the `name` field to the `name` of a data source defined in `DATA_SOURCES` in `config.json`.
```

>       * `mounted files`: If the `datatype` field is set to `mounted files`, the `name` field must be set to custom, and you should select custom tracks along with an optional bedfile.

Reviewer: We should backtick `custom`, and also `bedFile` (and capitalize it thus) if we are referring to the field. Also, we should have a noun phrase for what `mounted files` actually is.

Also, maybe we should say you need to "provide `tracks`" and not "select custom tracks".

>       * `examples`: Links to synthetic examples cannot currently be created.

Reviewer: If this can't be used, we should cut it from the bulleted list and the list of `datatype` options.

>    * Example: `dataType=built-in`

Reviewer: We should have another example showing the space in `mounted files` being encoded to give `mounted%20files`.

5. `Simplify`. 

>    This determines whether small snarls (e.g. single base indels and SNPs) are displayed or removed. It defaults to false, which means that the small snarls are not removed.

Reviewer: We might want to note here that it can't currently be used if there are any read tracks.

   * Example: `simplify=false`
6. `name` 

>    Name of Data. This is a field that indicates the name of preset data, which is defined in `DATA_SOURCES` in `config.json`. `name` is used when `datatype` is set to `built-in`. You do not have to use these presets. 

Reviewer: We should probably reiterate that this should be `custom` if `dataType` is `mounted files`.

   * Example: `name=snp1kg-BRCA1`

> 

Reviewer: It might be useful to provide one or two complete examples with all the fields filled in for a complete custom view with `mounted files` and just some non-default coordinates with `built-in`.

The key's name should **not** be in quotes.

###### Fields that can be included in the link URL:
1. `tracks` 

   Information about tracks. Tracks are objects consisting of `trackFile`, `trackType`, and `trackColorSettings`.

   A track object in JSON format might look like this:
   ```
   {
      "trackFile": "exampleData/internal/snp1kg-BRCA1.vg.xg",
      "trackType": "graph",
      "trackColorSettings": {
         "mainPalette": "greys",
         "auxPalette": "ygreys",
         "colorReadsByMappingQuality": false
      }
   }
   ```

   The collection of tracks wil be encoded as query string parameters starting with tracks.

   * The `trackFile` is the path (from the server working directory) or URL (any HTTP/HTTPS URL) to the track file. 
     * Examples:
         * File in server working directory: `tracks[0][trackFile]=exampleData/internal/snp1kg-BRCA1.vg.xg`.
         * HTTP/HTTPS URL: `tracks[0][trackFile]=https://public.gi.ucsc.edu/~anovak/graphs/trivial-brca1.vg`.
   * The `trackType` specifies the type of the track, which can be `graph`, `haplotype`, `read`.
      * Example: `tracks[0][trackType]=graph`.

>    * `trackColorSettings` (optional): the color information for the track. It has subfields:

Reviewer: These sibling bullets for `trackFile`, `trackType`, and `trackColorSettings` should all use the same sentence structure.

      * `mainPalette` and `auxPalette` (optional): The "main" and "auxilliary" color palettes to use, respectively. The values for each palette can be a hex code starting with a `#`, or one of the premade palette names: `greys`, `ygreys`, `blues`, `reds`, `plainColors`, or `lightColors`.
      * The palettes are used differently for different track types.
         * Graphs: the `mainPalette` colors the primary reference path while `auxPalette` is used for the other paths. 
         * Haplotypes: only the `mainPalette` is used. 
         * Reads: the `mainPalette` colors the forward-strand reads and the `auxPalette` colors the reverse-strand reads. 
         * Examples:
            * `tracks[0][trackColorSettings][mainPalette]=greys`.
            * `tracks[0][trackColorSettings][auxPalette]=ygreys`.
      * `colorReadsByMappingQuality` (optional) is a boolean value that determines if reads are colored based on their mapping quality or not. It defaults to `false`. 
         * Example: `tracks[0][trackColorSettings][colorReadsByMappingQuality]=true`.
   

2. `region`

>    Region coordinates. This is documented at step 3 of

Reviewer: "This" here sounds a little vague; we should maybe say that the syntax for specifying genomic coordinates is what is documented there.

>    [Displaying Visualizations](#displaying-visualizations). This region will be loaded in the tubemap visualization once the link is followed.

Reviewer: ```suggestion
   [Displaying Visualizations](#displaying-visualizations). This region will be loaded in the Tube Map visualization once the link is followed.
```

      * Example: `region=17%3A1-100`.

3. `bedFile` (optional)

   A BED file containing coordinates for regions of interest, some of which may have associated pre-extracted Tube Map views. More information on BED file structure and creation is documented [here](https://github.com/vgteam/sequenceTubeMap?tab=readme-ov-file#preparing-subgraphs-in-advance). 
   * Examples:
      * File in server working directory: `bedFile=exampleData/internal/snp1kg-BRCA1.bed`.
      * HTTP/HTTPS URL: `bedFile=https://raw.githubusercontent.com/vgteam/sequenceTubeMap/ca4f2485231ee4182173bec19489ba940b27461a/exampleData/cactus.bed`.
      * No BED file: `bedFile=none`.
   
4. `dataType`

   Information about the type of data. Data can be `built-in`, `mounted files`, or synthetic `examples`.
      * `built-in`: Predefined data sources from the server configuration file. If the `dataType` field is set to `built-in`, you should set the `name` field to the name of a data source defined in `DATA_SOURCES` in `config.json`.

>       * `mounted files`: If the `dataType` field is set to `mounted files`, the `name` field must be set to `custom`, and you should provide `custom` tracks along with an optional `bedFile`.

Reviewer: ```suggestion
      * `mounted files`: If the `dataType` field is set to `mounted files`, the `name` field must be set to `custom`, and you should provide custom tracks along with an optional `bedFile`.
```

      * Examples: 
         * `dataType=built-in`.
         * `dataType=mounted%20files`.

5. `Simplify`

   Whether small snarls (e.g. single base indels and SNPs) are displayed or removed. It defaults to false, which means that the small snarls are not removed. Currently, this option cannot be used if there are any read tracks.
      * Example: `simplify=false`.

6. `name` 

   Name of data. This is a field that indicates the name of preset data, which is defined in `DATA_SOURCES` in `config.json`. `name` is used when `dataType` is set to `built-in`. You do not have to use these presets. If `dataType` is `mounted files`, this field should be set to `custom`.
      * Examples:
         * `name=snp1kg-BRCA1`.

>          * `name=snp1kg-BRCA1`.

Reviewer: These are duplicated; maybe one is meant to be `custom`?
